lib/db/schema/patch-045-046.sql

```diff
@@ -0,0 +1,188 @@
+-- Add emails table and corresponding procedures to
+-- create, get and delete them.
+
+CREATE TABLE IF NOT EXISTS emails (
+  normalizedEmail VARCHAR(255) NOT NULL PRIMARY KEY,
+  email VARCHAR(255) NOT NULL,
+  uid BINARY(16) NOT NULL,
+  emailCode BINARY(16) NOT NULL,
+  isVerified BOOLEAN NOT NULL DEFAULT FALSE,
+  isPrimary BOOLEAN NOT NULL DEFAULT FALSE,
+  verifiedAt BIGINT UNSIGNED,
+  createdAt BIGINT UNSIGNED NOT NULL
+) ENGINE=InnoDB;
+
+DROP procedure IF EXISTS `createEmail_1`;
+
+CREATE PROCEDURE `createEmail_1` (
+    IN `normalizedEmail` VARCHAR(255),
+    IN `email` VARCHAR(255),
+    IN `uid` BINARY(16) ,
+    IN `emailCode` BINARY(16),
+    IN `isVerified` TINYINT(1),
+    IN `isPrimary` TINYINT(1),
+    IN `verifiedAt` BIGINT UNSIGNED,
+    IN `createdAt` BIGINT UNSIGNED
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+        BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    -- Currently, can not add an email that is specified in the
+    -- accounts table, regardless of verification state.
```

Comment:
> If it makes the code cleaner, I would not be opposed to duplicating the primary address both in the accounts table, and in this new emails table.

Do you mean inserting email in both tables? When would that happen, when the user attempts to add an additional email?

Ex. Flow
1) Checks to see if there is an email listed as primary in the emails table.
2) If not, inserts email on the account table into the emails table and sets it as primary for the account.
3) Inserts additional email and sets `isPrimary` to false.
4) All subsequent queries will only need to check the emails table to ensure uniqueness.